examples/step-66/step-66.cc

```diff
@@ -0,0 +1,1120 @@
+      std::string MPI_header = "Running with " + std::to_string(n_ranks) +
+                               " MPI process" + (n_ranks > 1 ? "es" : "");
+      std::string VEC_header =
+        "Vectorization over " + std::to_string(n_vect_doubles) +
+        " doubles = " + std::to_string(n_vect_bits) + " bits (" +
+        Utilities::System::get_current_vectorization_level() +
+        "), VECTORIZATION_LEVEL=" +
+        std::to_string(DEAL_II_COMPILER_VECTORIZATION_LEVEL);
+      std::string SOL_header = "Finite element space: " + fe.get_name();
+
+      pcout << std::string(80, '=') << std::endl;
+      pcout << DAT_header << std::endl;
+      pcout << std::string(80, '-') << std::endl;
+
+      pcout << MPI_header << std::endl;
+      pcout << VEC_header << std::endl;
+      pcout << SOL_header << std::endl;
+
+      pcout << std::string(80, '=') << std::endl;
+    }
+
+
+    for (unsigned int cycle = 0; cycle < 9 - dim; ++cycle)
+      {
+        pcout << std::string(80, '-') << std::endl;
+        pcout << "Cycle " << cycle << std::endl;
+        pcout << std::string(80, '-') << std::endl;
+
+
+        // The first task in actually solving the problem is to generate or
+        // refine the triangulation.
+        if (cycle == 0)
+          {
+            make_grid();
+          }
+        else
+          {
+            triangulation.refine_global(1);
+          }
+
+
+        // Now we set up the system and solve the problem. These steps are
+        // accompanied by time measurement and textual output.
+        Timer timer;
+
+        pcout << "Set up system..." << std::endl;
+        setup_system();
+
+        pcout << "   Triangulation: " << triangulation.n_global_active_cells()
+              << " cells" << std::endl;
+        pcout << "   DoFHandler:    " << dof_handler.n_dofs() << " DoFs"
+              << std::endl;
+        pcout << std::endl;
+
+
+        pcout << "Solve using Newton's method..." << std::endl;
+        solve();
+        pcout << std::endl;
+
+
+        timer.stop();
+        pcout << "Time for setup+solve (CPU/Wall) " << timer.cpu_time() << "/"
+              << timer.wall_time() << " s" << std::endl;
+        pcout << std::endl;
+
+
+        // After the problem was solved we compute the norm of the solution
+        // and generate the graphical output files.
+        pcout << "Output results..." << std::endl;
+        const double norm = compute_solution_norm();
+        output_results(cycle);
+
+        pcout << "  H1 seminorm: " << norm << std::endl;
+        pcout << std::endl;
+
+
+        // Finally after each cycle we print the timing information.
+        computing_timer.print_summary();
+        computing_timer.reset();
+      }
+  }
+} // namespace Step66
+
+
+
+// @sect3{The main function}
+
+// As typical for programs running in parallel with MPI we set up the MPI
+// framework and limit the number of threads to one. Finally to run the solver
```

Comment:
```
// framework. Finally to run the solver
```

I would not be surprised that the code also works with threads. Could you try it out. Thx.

Reply:
Of course I will try it. BUt maybe let us postpone this detail to a later update of the example, as time is running?

Reply:
Sure.